base_layer/core/src/transactions/transaction_components/encrypted_data.rs

```diff
@@ -117,12 +195,19 @@ impl EncryptedData {
         encryption_key: &PrivateKey,
         commitment: &Commitment,
         encrypted_data: &EncryptedData,
-    ) -> Result<(MicroMinotari, PrivateKey), EncryptedDataError> {
+    ) -> Result<(MicroMinotari, PrivateKey, PaymentId), EncryptedDataError> {
         // Extract the nonce, ciphertext, and tag
-        let nonce = XNonce::from_slice(&encrypted_data.as_bytes()[..SIZE_NONCE]);
-        let mut bytes = Zeroizing::new([0u8; SIZE_VALUE + SIZE_MASK]);
-        bytes.clone_from_slice(&encrypted_data.as_bytes()[SIZE_NONCE..SIZE_NONCE + SIZE_VALUE + SIZE_MASK]);
-        let tag = Tag::from_slice(&encrypted_data.as_bytes()[SIZE_NONCE + SIZE_VALUE + SIZE_MASK..]);
+        let tag = Tag::from_slice(&encrypted_data.as_bytes()[..SIZE_TAG]);
```

Comment:
Watch out for buffer overrun. Possibly fuzz this with quicktest

Reply:
those should all be a constant size